ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/SciRouterOutlet.java

```diff
@@ -120,16 +127,22 @@ private IDisposable installRouter(final String outletName) {
       }
     });
 
-    return () -> subscription.unsubscribe();
+    return subscription::unsubscribe;
   }
 
-  private IDisposable installKeystrokeDispatcher() {
+  private IDisposable installKeystrokeDispatcher(final Control keystrokeTarget) {
+    if (keystrokeTarget == null) {
+      return () -> {
+        // noop
+      };
+    }
     return routerOutletProxy.onKeystroke(event -> {
-      Platform.getLog(SciRouterOutlet.class).info(String.format(//
-          "TODO: Dispatching event to SWT [type=%s, keyCode=%s, character=%s, stateMask=%s]", //
-          Integer.valueOf(event.type), Integer.valueOf(event.keyCode), Character.valueOf(event.character),
-          Integer.valueOf(event.stateMask)));
-      // TODO Dispatching event to SWT (getDisplay().post(event))
+      keystrokeTarget.setFocus();

```

Comment:
Won't this introduce another issue if the browser loses the focus for each propagated keystroke, if for example the keystroke happens inside a textbox?